From 5deee6f9ada543578d1289467352126c52ebca41 Mon Sep 17 00:00:00 2001
From: Peter De Schrijver <pdeschrijver@nvidia.com>
Date: Thu, 12 Jun 2014 18:36:36 +0300
Subject: [PATCH 17/45] ARM: tegra: move fuse exports to soc/tegra/fuse.h

All fuse related functionality will move to a driver in the following
patches. To prepare for this, export all the required functionality in a
global header file and move all users of fuse.h to soc/tegra/fuse.h.

While we're at it, remove tegra_bct_strapping, as its only user was
removed in Commit a7cbe92cef27 ("ARM: tegra: remove tegra EMC scaling
driver").

Signed-off-by: Peter De Schrijver <pdeschrijver@nvidia.com>
Signed-off-by: Stephen Warren <swarren@nvidia.com>
Signed-off-by: Thierry Reding <treding@nvidia.com>
---
 arch/arm/mach-tegra/fuse.h            | 13 -------------
 arch/arm/mach-tegra/tegra.c           |  1 -
 arch/arm/mach-tegra/tegra114_speedo.c |  2 ++
 arch/arm/mach-tegra/tegra20_speedo.c  |  2 ++
 arch/arm/mach-tegra/tegra30_speedo.c  |  2 ++
 include/soc/tegra/fuse.h              | 16 ++++++++++++++++
 6 files changed, 22 insertions(+), 14 deletions(-)

diff --git a/arch/arm/mach-tegra/fuse.h b/arch/arm/mach-tegra/fuse.h
index 7a08b4b..48a4886 100644
--- a/arch/arm/mach-tegra/fuse.h
+++ b/arch/arm/mach-tegra/fuse.h
@@ -27,27 +27,14 @@
 #define SKU_ID_T25E	28
 
 #ifndef __ASSEMBLY__
-enum tegra_revision {
-	TEGRA_REVISION_UNKNOWN = 0,
-	TEGRA_REVISION_A01,
-	TEGRA_REVISION_A02,
-	TEGRA_REVISION_A03,
-	TEGRA_REVISION_A03p,
-	TEGRA_REVISION_A04,
-	TEGRA_REVISION_MAX,
-};
 
 extern int tegra_sku_id;
 extern int tegra_cpu_process_id;
 extern int tegra_core_process_id;
 extern int tegra_cpu_speedo_id;		/* only exist in Tegra30 and later */
 extern int tegra_soc_speedo_id;
-extern enum tegra_revision tegra_revision;
-
-extern int tegra_bct_strapping;
 
 unsigned long long tegra_chip_uid(void);
-void tegra_init_fuse(void);
 bool tegra_spare_fuse(int bit);
 u32 tegra_fuse_readl(unsigned long offset);
 
diff --git a/arch/arm/mach-tegra/tegra.c b/arch/arm/mach-tegra/tegra.c
index 8be25c4..a359931 100644
--- a/arch/arm/mach-tegra/tegra.c
+++ b/arch/arm/mach-tegra/tegra.c
@@ -48,7 +48,6 @@
 #include "board.h"
 #include "common.h"
 #include "cpuidle.h"
-#include "fuse.h"
 #include "iomap.h"
 #include "irq.h"
 #include "pmc.h"
diff --git a/arch/arm/mach-tegra/tegra114_speedo.c b/arch/arm/mach-tegra/tegra114_speedo.c
index 86eca17..d0a6d59 100644
--- a/arch/arm/mach-tegra/tegra114_speedo.c
+++ b/arch/arm/mach-tegra/tegra114_speedo.c
@@ -17,6 +17,8 @@
 #include <linux/bug.h>
 #include <linux/kernel.h>
 
+#include <soc/tegra/fuse.h>
+
 #include "fuse.h"
 
 #define CORE_PROCESS_CORNERS_NUM	2
diff --git a/arch/arm/mach-tegra/tegra20_speedo.c b/arch/arm/mach-tegra/tegra20_speedo.c
index dcd29a0..2907cf8 100644
--- a/arch/arm/mach-tegra/tegra20_speedo.c
+++ b/arch/arm/mach-tegra/tegra20_speedo.c
@@ -17,6 +17,8 @@
 #include <linux/bug.h>
 #include <linux/kernel.h>
 
+#include <soc/tegra/fuse.h>
+
 #include "fuse.h"
 
 #define CPU_SPEEDO_LSBIT		20
diff --git a/arch/arm/mach-tegra/tegra30_speedo.c b/arch/arm/mach-tegra/tegra30_speedo.c
index 7c00383..6f6102c 100644
--- a/arch/arm/mach-tegra/tegra30_speedo.c
+++ b/arch/arm/mach-tegra/tegra30_speedo.c
@@ -17,6 +17,8 @@
 #include <linux/bug.h>
 #include <linux/kernel.h>
 
+#include <soc/tegra/fuse.h>
+
 #include "fuse.h"
 
 #define CORE_PROCESS_CORNERS_NUM	1
diff --git a/include/soc/tegra/fuse.h b/include/soc/tegra/fuse.h
index a43a750..822eb34 100644
--- a/include/soc/tegra/fuse.h
+++ b/include/soc/tegra/fuse.h
@@ -27,6 +27,22 @@
 u32 tegra_read_chipid(void);
 u8 tegra_get_chip_id(void);
 
+enum tegra_revision {
+	TEGRA_REVISION_UNKNOWN = 0,
+	TEGRA_REVISION_A01,
+	TEGRA_REVISION_A02,
+	TEGRA_REVISION_A03,
+	TEGRA_REVISION_A03p,
+	TEGRA_REVISION_A04,
+	TEGRA_REVISION_MAX,
+};
+
+u32 tegra_read_straps(void);
+u32 tegra_read_chipid(void);
+void tegra_init_fuse(void);
+
+extern enum tegra_revision tegra_revision;
+
 #if defined(CONFIG_TEGRA20_APB_DMA)
 int tegra_apb_readl_using_dma(unsigned long offset, u32 *value);
 int tegra_apb_writel_using_dma(u32 value, unsigned long offset);
-- 
2.0.1

